ast/compile.go

```diff
@@ -228,6 +228,9 @@ type QueryCompiler interface {
 	// ComprehensionIndex returns an index data structure for the given comprehension
 	// term. If no index is found, returns nil.
 	ComprehensionIndex(term *Term) *ComprehensionIndex
+
+	// WithStrict enables strict mode for the query compiler.
+	WithStrict(yes bool) QueryCompiler
```

Comment:
I wonder if we need this at this point? Alternatively, we could set `c0.strict = false` in the QueryCompiler() method below, https://github.com/open-policy-agent/opa/pull/5228/files#diff-067a842cc5e1a348551f4c7656120105acedafb2faac6e7b7ff57b75c5349570R426

Reply:
Yeah, that sounds better indeed, I will try this :+1:

Reply:
Potentially a dumb question, but is this not possible? :thinking: 

```
func (c *Compiler) QueryCompiler() QueryCompiler {
	c.init()
	c0 := *c
	c0.strict = false
	return newQueryCompiler(&c0)
}
```

Reply:
Sorry you've lost me -- this is what I had in mind. Is it possible?

Reply:
Yeah we are on the same page :) - https://github.com/open-policy-agent/opa/pull/5228/files#diff-067a842cc5e1a348551f4c7656120105acedafb2faac6e7b7ff57b75c5349570R426 took me to 

```
func (qc *queryCompiler) WithStrict(yes bool) QueryCompiler {
	qc.compiler.WithStrict(false)
	return qc
}
```

Which is where I was confused so I can remove all that and it will look good!

I will tidy up :broom: and push again in a bit!

Thanks again for your great advice!